### PR TITLE
feat: support to set icon color to the theme color when proxy icon starts with `data:image/svg+xml`

### PR DESCRIPTION
**Pull request by lamprose**
image

### CODE REVIEW

File: src/pages/Proxies.tsx [hunk 1]
```diff
@@ -201,6 +221,26 @@ export default () => {
...
+                          <Show
+                            when={
+                              proxyGroup.icon &&
+                              proxyGroup.icon.startsWith('data:image/svg+xml')
+                            }
+                          >
+                            <div
+                              style={{
+                                'mask-image': `url("${encodeSvg(proxyGroup.icon!)}")`,
+                                '-webkitmask-image': `url("${encodeSvg(proxyGroup.icon!)}")`,
```
Comment (kunish) [inline src/pages/Proxies.tsx:233]: First, you might want to use `-webkit-mask-image`? Second, I couldn't find this property anywhere.

image

File: src/pages/Proxies.tsx [hunk 2]
```diff
@@ -201,6 +221,26 @@ export default () => {
...
+                                height: `${iconHeight()}px`,
+                                width: `${iconHeight()}px`,
+                                'margin-right': `${iconMarginRight()}px`,
+                                'mask-size': '100% 100%',
```
Comment (kunish) [inline src/pages/Proxies.tsx:237]: Rearrange these properties
```suggestion
                                'mask-image': `url("${encodeSvg(proxyGroup.icon!)}")`,
                                'mask-size': '100% 100%',
```

File: src/pages/Proxies.tsx [hunk 3]
```diff
@@ -201,6 +221,26 @@ export default () => {
...
+                                'background-color': 'currentColor',
+                                color:
+                                  'var(--icon-p, oklch(var(--p) / var(--tw-bg-opacity)))',
```
Comment (kunish) [inline src/pages/Proxies.tsx:240]: Where are these css variables come from?

File: src/pages/Proxies.tsx [hunk 4]
```diff
@@ -92,7 +92,22 @@ export default () => {
     e.stopPropagation()
     void proxyGroupLatencyTest(groupName)
   }
-
+  const encodeSvg = (svg: string) => {
```
Comment (kunish) [inline src/pages/Proxies.tsx:95]: What does this do? Add some comments and examples to illustrate please.

File: src/pages/Proxies.tsx [hunk 5]
```diff
@@ -192,7 +207,12 @@ export default () => {
                     <div class="space-y-2">
                       <div class="flex items-center justify-between pr-8">
                         <div class="flex items-center">
-                          <Show when={proxyGroup.icon}>
+                          <Show
```
Comment (kunish) [inline src/pages/Proxies.tsx:210]: If you want to replicate `Ternary Operator` in jsx with `Show` component, you can put `else` case component in `fallback` property. See https://docs.solidjs.com/reference/components/show

File: src/pages/Proxies.tsx [hunk 6]
```diff
@@ -192,7 +207,12 @@ export default () => {
...
+                            when={
+                              proxyGroup.icon &&
```
Comment (kunish) [inline src/pages/Proxies.tsx:212]: Also take advantage of javascript feature [Optional Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

**Review by kunish — CHANGES_REQUESTED** [review]

File: src/pages/Proxies.tsx (hunk 1 above)
Comment (lamprose) [reply src/pages/Proxies.tsx]: it's incorrect CSS properties,delete it

File: src/pages/Proxies.tsx (hunk 2 above)
Comment (lamprose) [reply src/pages/Proxies.tsx]: Done

File: src/pages/Proxies.tsx (hunk 4 above)
Comment (lamprose) [reply src/pages/Proxies.tsx:99]: transform an SVG into a data URI
,the function referred to `https://bl.ocks.org/jennyknuth/222825e315d45a738ed9d6e04c7a88d0`

File: src/pages/Proxies.tsx (hunk 5 above)
Comment (lamprose) [reply src/pages/Proxies.tsx]: Done

File: src/pages/Proxies.tsx (hunk 6 above)
Comment (lamprose) [reply src/pages/Proxies.tsx]: Done

File: src/pages/Proxies.tsx (hunk 3 above)
Comment (lamprose) [reply src/pages/Proxies.tsx]: the new commit has already deleted the css variables

File: src/pages/Proxies.tsx [hunk 7]
```diff
@@ -193,13 +212,33 @@ export default () => {
...
                           <Show when={proxyGroup.icon}>
-                            <img
-                              src={proxyGroup.icon}
-                              style={{
-                                height: `${iconHeight()}px`,
-                                'margin-right': `${iconMarginRight()}px`,
-                              }}
-                            />
+                            <Show
+                              when={proxyGroup.icon?.startsWith(
```
Comment (kunish) [inline src/pages/Proxies.tsx:216]:
```suggestion
                              when={proxyGroup.icon!.startsWith(
```

**Review by kunish — CHANGES_REQUESTED** [review]

**Pull request merged**